### PR TITLE
[GLUTEN-7174][VL] Force fallback scan operator when spark.sql.parquet.mergeSchema enabled

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Velox treat missing column as null, will output incorrect results when spark.sql.parquet.mergeSchema enabled.

Fixes: #7174

## How was this patch tested?

Enable some excluded tests.

### CODE REVIEW

**Review by zhztheplayer — COMMENTED** [review]
I see test `Gluten - Filter applied on merged Parquet schema with new column should work` is removed from 3.3 / 3.4 / 3.5 UTs but is preserved in 3.2 UT. Would you like to explain a little bit? Thanks.

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala [hunk 1]
```diff
@@ -88,78 +88,92 @@ object VeloxBackendSettings extends BackendSettingsApi {
...
           }
-          validateTypes(typeValidator)
-        }
-      case _ => ValidationResult.failed(s"Unsupported file format for $format.")
+          if (SQLConf.get.isParquetSchemaMergingEnabled) {
+            // https://github.com/apache/incubator-gluten/issues/7174
+            Some(s"not support when ${SQLConf.PARQUET_SCHEMA_MERGING_ENABLED.key} is true")
+          } else {
+            validateTypes(typeValidator)
+          }
```
Comment (zhztheplayer) [inline backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:143]: By looking into [this Spark code](https://github.com/apache/spark/blob/51e915a0ca2d3f1c9d8337d1ee9692247026c40f/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala#L69-L72), I assume the config `SQLConf.PARQUET_SCHEMA_MERGING_ENABLED.key` can be overridden dynamically in file read options? Can we use the final overridden value of `ParquetOptions.mergeSchema`?

**Comment by Yohahaha** [comment]
> I see test `Gluten - Filter applied on merged Parquet schema with new column should work` is removed from 3.3 / 3.4 / 3.5 UTs but is preserved in 3.2 UT. Would you like to explain a little bit? Thanks.

oh, I forget the 3.2 UT was still running.

**Review by zhztheplayer — APPROVED** [review]
👍

**Comment by zhztheplayer** [comment]
If `Gluten - Filter applied on merged Parquet schema with new column should work` is removed, can we directly use vanilla Spark's `Filter applied on merged Parquet schema with new column should work` once we got the incompatibility fixed?

**Comment by Yohahaha** [comment]
> If `Gluten - Filter applied on merged Parquet schema with new column should work` is removed, can we directly use vanilla Spark's `Filter applied on merged Parquet schema with new column should work` once we got the incompatibility fixed?

yes, just run `Filter applied on merged Parquet schema with new column should work` and remove the merged schema check.

**Comment by FelixYBW** [comment]
Is it common? should we fix it in Velox in future?

**Comment by Yohahaha** [comment]
> Is it common? should we fix it in Velox in future?

I think it's not common.

**Comment by Yohahaha** [comment]
```
[2024-10-29T03:36:06.075Z] Run completed in 22 minutes, 8 seconds.
[2024-10-29T03:36:06.075Z] Total number of tests run: 5482
[2024-10-29T03:36:06.075Z] Suites: completed 227, aborted 0
[2024-10-29T03:36:06.075Z] Tests: succeeded 5482, failed 0, canceled 0, ignored 2502, pending 0
[2024-10-29T03:36:06.075Z] All tests passed.
[2024-10-29T03:36:07.440Z] [INFO] ------------------------------------------------------------------------
[2024-10-29T03:36:07.440Z] [INFO] Reactor Summary for Gluten Parent Pom 1.3.0-SNAPSHOT:
[2024-10-29T03:36:07.440Z] [INFO] 
[2024-10-29T03:36:07.440Z] [INFO] Gluten Parent Pom .................................. SUCCESS [  0.744 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Ras ......................................... SUCCESS [  7.920 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Ras Common .................................. SUCCESS [ 39.893 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Shims ....................................... SUCCESS [  1.596 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Shims Common ................................ SUCCESS [ 11.082 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Shims for Spark 3.2 ......................... SUCCESS [ 15.003 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten UI .......................................... SUCCESS [  3.975 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Core ........................................ SUCCESS [ 18.820 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Substrait ................................... SUCCESS [ 42.338 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Backends ClickHouse ......................... SUCCESS [51:17 min]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Celeborn .................................... SUCCESS [  0.292 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Celeborn Common ............................. SUCCESS [  3.512 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Celeborn Clickhouse ......................... SUCCESS [01:25 min]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Celeborn Package ............................ SUCCESS [  0.319 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Package ..................................... SUCCESS [  2.723 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Ras Planner ................................. SUCCESS [  1.092 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Unit Test Parent ............................ SUCCESS [  0.347 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Unit Test Common ............................ SUCCESS [  7.357 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Unit Test ................................... SUCCESS [ 12.338 s]
[2024-10-29T03:36:07.440Z] [INFO] Gluten Unit Test Spark32 ........................... SUCCESS [22:28 min]
[2024-10-29T03:36:07.440Z] [INFO] ------------------------------------------------------------------------
[2024-10-29T03:36:07.440Z] [INFO] BUILD SUCCESS
[2024-10-29T03:36:07.440Z] [INFO] ------------------------------------------------------------------------
[2024-10-29T03:36:07.440Z] [INFO] Total time:  01:18 h
[2024-10-29T03:36:07.440Z] [INFO] Finished at: 2024-10-29T03:36:07Z
[2024-10-29T03:36:07.440Z] [INFO] ------------------------------------------------------------------------
```

CH's CI is success indeed.

**Comment by Yohahaha** [comment]
Run Gluten Clickhouse CI

**Comment by Yohahaha** [comment]
@baibaichen @zzcclp could you help double check whether the CH CI is passed? seems it's still give incorrect result.

**Comment by Yohahaha** [comment]
Run Gluten Clickhouse CI

**Pull request merged**